### PR TITLE
Sensor data structure : clean all sensor data from common include files

### PR DESCRIPTION
**Pull request by MaciekWronski**
Reorganise top level sensor module and clean sensor data from common files

#### Description of the changes

I have included NSENSOR variable into the top level sensor structure and deleted it from com04 include file.

In the same way, fixed lenght arrays containing list of sensors defined in engine input options were deleted from other common include files and added in the principal sensor data structure as allocatables

### CODE REVIEW

**Review by laurent-altr — CHANGES_REQUESTED** [review]
Hello @MaciekWronski, 

Could you please review the commit message: 
https://openpbs.atlassian.net/wiki/spaces/DG/pages/6193155/How+To+Write+a+Good+Git+Commit+Message

File: engine/source/mpi/interfaces/spmd_ifront.F [hunk 1]
```diff
@@ -74,7 +74,7 @@ SUBROUTINE SPMD_IFRONT(
 C-----------------------------------------------
       INTEGER NBINTC,ISLEN7,IRLEN7,ISLEN11,IRLEN11,ISLEN17,IRLEN17,
      .        IRLEN7T,ISLEN7T,IRLEN20,ISLEN20 ,IRLEN20T,ISLEN20T,
-     .        IRLEN20E,ISLEN20E,  
+     .        IRLEN20E,ISLEN20E, NSENSOR,
```
Comment (sebastienVilleneuve) [inline engine/source/mpi/interfaces/spmd_ifront.F:77]: INTENT statement is missing

File: engine/source/output/h3d/h3d_build_fortran/h3d_read.F [hunk 2]
```diff
@@ -54,6 +55,7 @@ SUBROUTINE H3D_READ(IKAD,KEY0,KH3D,NSLASH,H3D_DATA)
       INTEGER IKAD(0:*),KH3D,NSLASH(*)
       CHARACTER KEY0(*)*5
       TYPE (H3D_DATABASE) :: H3D_DATA
+      TYPE (SENSORS_)     :: SENSORS
```
Comment (sebastienVilleneuve) [inline engine/source/output/h3d/h3d_build_fortran/h3d_read.F:58]: INTENT statement is missing

File: engine/source/output/report/report.F [hunk 3]
```diff
@@ -504,10 +505,10 @@ SUBROUTINE REPORT(IH3DREADER,N,IMVW)
          WRITE(IOUTREP,'(A,I10,A)')'<td><font face="Arial" size="2">',NSLIN,'</td>'
          WRITE(IOUTREP,'(A)')'</tr>'
        ENDIF
-       IF(NSENSOR /= 0)THEN
+       IF(SENSORS%NSENSOR /= 0)THEN
```
Comment (sebastienVilleneuve) [inline engine/source/output/report/report.F:508]: SENSORS not passed by argument

File: engine/source/output/restart/read_sensors.F [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+C-----------------------------------------------
+#include "implicit_f.inc"
+#include "scr07_c.inc"
+#include "scr17_c.inc"
+#include "nchar_c.inc"
+#include "my_allocate.inc"
+C-----------------------------------------------
+C   D u m m y   A r g u m e n t s
+C-----------------------------------------------
+      TYPE (SENSORS_) ,TARGET :: SENSORS
```
Comment (sebastienVilleneuve) [inline engine/source/output/restart/read_sensors.F:50]: INTENT statement is missing

File: engine/source/output/sortie_main.F [hunk 5]
```diff
@@ -259,7 +259,7 @@ SUBROUTINE SORTIE_MAIN(
       TYPE (PINCH) :: PINCH_DATA
       TYPE (DRAPE_)  :: DRAPE_SH4N(NUMELC_DRAPE), DRAPE_SH3N(NUMELTG_DRAPE)
       TYPE (DRAPEG_)  :: DRAPEG
-      TYPE (SENSOR_STR_), DIMENSION(NSENSOR) :: SENSOR_TAB
+      TYPE (SENSORS_) :: SENSORS
```
Comment (sebastienVilleneuve) [inline engine/source/output/sortie_main.F:262]: INTENT statement is missing

File: starter/source/user_interface/uaccess.F [hunk 6]
```diff
@@ -934,11 +934,11 @@ INTEGER FUNCTION SET_U_SENS_IPAR(IVAR,VAR)
         RETURN
       ENDIF
 C
-      ITYP = SENSOR_TAB(NSENS)%TYPE
+      ITYP = SENSORS%SENSOR_TAB(NSENS)%TYPE
```
Comment (sebastienVilleneuve) [inline starter/source/user_interface/uaccess.F:937]: SENSOR not passed by argument

File: starter/source/tools/sensor/sort_logical_sensors.F [hunk 7]
```diff
@@ -66,6 +66,7 @@ SUBROUTINE SORT_LOGICAL_SENSORS(SENSOR_TAB)
 C-----------------------------------------------
 C   D u m m y   A r g u m e n t s
 C-----------------------------------------------
+      INTEGER NSENSOR
```
Comment (sebastienVilleneuve) [inline starter/source/tools/sensor/sort_logical_sensors.F:69]: INTENT statement is missing

File: starter/source/tools/sensor/sensor_tab_init.F [hunk 8]
```diff
@@ -39,27 +39,24 @@ SUBROUTINE SENSOR_TAB_INIT(SENSOR_TAB,NSENSOR)
 C-----------------------------------------------
 C   D u m m y   A r g u m e n t s
 C-----------------------------------------------
-      INTEGER NSENSOR
-      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR), TARGET :: SENSOR_TAB
+      TYPE (SENSORS_) :: SENSORS
```
Comment (sebastienVilleneuve) [inline starter/source/tools/sensor/sensor_tab_init.F:42]: INTENT statement is missing

File: starter/source/tools/sensor/inisen.F [hunk 9]
```diff
@@ -51,6 +51,7 @@ SUBROUTINE INISEN(SENSOR_TAB, IPARI ,NOM_OPT,PTR_NOPT_RWALL,
 C-----------------------------------------------
 C   D u m m y   A r g u m e n t s
 C-----------------------------------------------
+      INTEGER NSENSOR
```
Comment (sebastienVilleneuve) [inline starter/source/tools/sensor/inisen.F:54]: INTENT statement is missing

File: starter/source/tools/sensor/hm_read_sensors.F [hunk 10]
```diff
@@ -98,7 +99,7 @@ SUBROUTINE HM_READ_SENSORS(
       INTEGER ,DIMENSION(LIPART1,*) ,INTENT(IN) :: IPART
       my_real ,DIMENSION(LSKEW,*) :: SKEW
       my_real ,DIMENSION(*)       :: BUFSF
-      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR), TARGET :: SENSOR_TAB
+      TYPE (SENSORS_)  ,TARGET    :: SENSORS
```
Comment (sebastienVilleneuve) [inline starter/source/tools/sensor/hm_read_sensors.F:102]: INTENT statement is missing

File: starter/source/tools/seatbelts/create_seatbelt.F [hunk 11]
```diff
@@ -61,7 +61,7 @@ SUBROUTINE CREATE_SEATBELT(IXR,ITAB,KNOD2EL1D,NOD2EL1D,IPM,
      .        KNOD2ELC(*),NOD2ELC(*),IXC(NIXC,*)
       INTEGER, INTENT(INOUT) :: IGEO(NPROPGI,NUMGEO),ISKN(SISKWN)
       my_real X(3,*),BUFMAT(*),PM(NPROPM,*),GEO(NPROPG,*)
-      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) :: SENSOR_TAB
+      TYPE (SENSORS_) :: SENSORS
```
Comment (sebastienVilleneuve) [inline starter/source/tools/seatbelts/create_seatbelt.F:64]: INTENT statement is missing

File: starter/source/system/fsdcod.F [hunk 12]
```diff
@@ -81,7 +81,7 @@ SUBROUTINE FSDCOD(BUFMAT    ,PM        ,GEO       ,IBCL      ,IPRES   ,
       TYPE(TTABLE) , DIMENSION(NTABLE) :: TABLE
       DOUBLE PRECISION BUFGEO(*)
       INTEGER NOM_OPT(LNOPT1,*)
-      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) :: SENSOR_TAB
+      TYPE (SENSORS_)  :: SENSORS
```
Comment (sebastienVilleneuve) [inline starter/source/system/fsdcod.F:84]: INTENT statement is missing

File: starter/source/starter/lectur.F [hunk 13]
```diff
@@ -3751,7 +3753,7 @@ SUBROUTINE LECTUR( MULTI_FVM                   ,LSUBMODEL          ,IS_DYNA
      5           STACK%PM ,IWORKSH  ,IGRBRIC ,IGRQUAD   ,IGRSH4N ,
      6           IGRSH3N  ,IGRTRUSS ,IGRBEAM ,IGRSPRING ,IGRNOD  ,
      7           IGRSURF  ,IGRSLIN, LSUBMODEL,ALE_EULER ,IGEO    ,
-     8           NLOC_DMG ,DETONATORS)
+     8           NLOC_DMG ,DETONATORS,SENSORS%NSENSOR)
```
Comment (sebastienVilleneuve) [inline starter/source/starter/lectur.F:3756]: SENSOR not passed by argument ( no global variable allowed)

File: starter/source/restart/ddsplit/w_front.F [hunk 14]
```diff
@@ -98,7 +98,7 @@ SUBROUTINE W_FRONT(
      .          NESVOIS, NSEGFL_L, NUMEL, NBCFD, NUMELQ_L,NUMELTG_L,
      .          NUMPOR_L, NUMEL_L, NBI18_L, INACTI, NLAGF_L,
      .          NBDDNORT, NBDDNOR_MAX, NBCCFR25, NBCCNOR, NUMNOR_L,
-     .          NBDDEDGT,NBDDEDG_MAX,
+     .          NBDDEDGT,NBDDEDG_MAX,NSENSOR,
```
Comment (sebastienVilleneuve) [inline starter/source/restart/ddsplit/w_front.F:101]: INTENT statement is missing

File: engine/source/tools/sensor/sensor_dist_surf0.F [hunk 15]
```diff
@@ -59,9 +59,10 @@ SUBROUTINE SENSOR_DIST_SURF0(SENSOR_PTR,X,IGRSURF)
 C-----------------------------------------------
 C   D u m m y   A r g u m e n t s
 C-----------------------------------------------
+      INTEGER ,INTENT(IN) :: NSENSOR
       my_real ,DIMENSION(3,NUMNOD) :: X
       TYPE (SURF_) ,DIMENSION(NSURF), TARGET :: IGRSURF
-      TYPE (SENSOR_STR_), DIMENSION(NSENSOR) :: SENSOR_PTR
+      TYPE (SENSOR_STR_), DIMENSION(NSENSOR) :: SENSOR_TAB
```
Comment (sebastienVilleneuve) [inline engine/source/tools/sensor/sensor_dist_surf0.F:65]: INTENT statement is missing

File: engine/source/user_interface/usensor.F [hunk 16]
```diff
@@ -67,10 +67,10 @@ INTEGER FUNCTION GET_U_NUMSENS(IDSENS)
         RETURN
       ENDIF
 C---
-      DO J=1,NSENSOR
+      DO J=1,SENSORS%NSENSOR
```
Comment (sebastienVilleneuve) [inline engine/source/user_interface/usensor.F:70]: SENSOR not passed by arguments

File: starter/source/airbag/hm_read_monvol_type11.F [hunk 17]
```diff
@@ -90,7 +90,7 @@ SUBROUTINE HM_READ_MONVOL_TYPE11(T_MONVOLN, IPM, IGEO, ITABM1,
       TYPE(MONVOL_STRUCT_), INTENT(INOUT) :: T_MONVOLN
       TYPE (GROUP_)  , DIMENSION(NGRBRIC), INTENT(IN) :: IGRBRIC
       TYPE(SUBMODEL_DATA), DIMENSION(NSUBMOD), INTENT(IN) :: LSUBMODEL
-      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) :: SENSOR_TAB
+      TYPE (SENSORS_)  :: SENSORS
```
Comment (sebastienVilleneuve) [inline starter/source/airbag/hm_read_monvol_type11.F:93]: INTENT statement is missing

File: starter/source/airbag/hm_read_monvol_type5.F [hunk 18]
```diff
@@ -88,7 +88,7 @@ SUBROUTINE HM_READ_MONVOL_TYPE5(T_MONVOLN, T_MONVOL_METADATA, IPM, IGEO, ITABM1,
       TYPE(MONVOL_STRUCT_), INTENT(INOUT) :: T_MONVOLN
       TYPE(MONVOL_METADATA_), INTENT(INOUT) :: T_MONVOL_METADATA
       TYPE(SUBMODEL_DATA), DIMENSION(NSUBMOD), INTENT(IN) :: LSUBMODEL
-      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) :: SENSOR_TAB
+      TYPE (SENSORS_)  :: SENSORS
```
Comment (sebastienVilleneuve) [inline starter/source/airbag/hm_read_monvol_type5.F:91]: INTENT statement is missing

File: starter/source/airbag/hm_read_monvol_type6.F [hunk 19]
```diff
@@ -87,7 +87,7 @@ SUBROUTINE HM_READ_MONVOL_TYPE6(T_MONVOLN, IPM, IGEO, ITABM1,
       TYPE(MONVOL_STRUCT_), INTENT(INOUT) :: T_MONVOLN
       TYPE (GROUP_), DIMENSION(NGRBRIC), INTENT(IN) :: IGRBRIC
       TYPE(SUBMODEL_DATA), DIMENSION(NSUBMOD), INTENT(IN) :: LSUBMODEL
-      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) :: SENSOR_TAB
+      TYPE (SENSORS_)  :: SENSORS
```
Comment (sebastienVilleneuve) [inline starter/source/airbag/hm_read_monvol_type6.F:90]: INTENT statement is missing

File: starter/source/airbag/hm_read_monvol_type7.F [hunk 20]
```diff
@@ -90,7 +90,7 @@ SUBROUTINE HM_READ_MONVOL_TYPE7(T_MONVOLN, IPM, IGEO, ITABM1,
       TYPE (SURF_), INTENT(INOUT), DIMENSION(NSURF) :: IGRSURF
       TYPE(MONVOL_STRUCT_), INTENT(INOUT) :: T_MONVOLN
       TYPE(SUBMODEL_DATA), DIMENSION(NSUBMOD), INTENT(IN) :: LSUBMODEL
-      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) :: SENSOR_TAB
+      TYPE (SENSORS_)  :: SENSORS
```
Comment (sebastienVilleneuve) [inline starter/source/airbag/hm_read_monvol_type7.F:93]: INTENT statement is missing

File: starter/source/airbag/hm_read_monvol_type8.F [hunk 21]
```diff
@@ -91,7 +91,7 @@ SUBROUTINE HM_READ_MONVOL_TYPE8(T_MONVOLN, IPM, IGEO, ITABM1,
       TYPE(MONVOL_STRUCT_), INTENT(INOUT) :: T_MONVOLN
       TYPE (GROUP_)  , DIMENSION(NGRBRIC), INTENT(IN) :: IGRBRIC
       TYPE(SUBMODEL_DATA), DIMENSION(NSUBMOD), INTENT(IN) :: LSUBMODEL
-      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) :: SENSOR_TAB
+      TYPE (SENSORS_)  :: SENSORS
```
Comment (sebastienVilleneuve) [inline starter/source/airbag/hm_read_monvol_type8.F:94]: INTENT statement is missing

File: starter/source/airbag/hm_read_monvol_type9.F [hunk 22]
```diff
@@ -90,7 +90,7 @@ SUBROUTINE HM_READ_MONVOL_TYPE9(T_MONVOLN, T_MONVOL_METADATA, IPM, IGEO, ITABM1,
       TYPE(MONVOL_STRUCT_), INTENT(INOUT) :: T_MONVOLN
       TYPE(MONVOL_METADATA_), INTENT(INOUT) :: T_MONVOL_METADATA
       TYPE(SUBMODEL_DATA), DIMENSION(NSUBMOD), INTENT(IN) :: LSUBMODEL
-      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) :: SENSOR_TAB
+      TYPE (SENSORS_)  :: SENSORS
```
Comment (sebastienVilleneuve) [inline starter/source/airbag/hm_read_monvol_type9.F:93]: INTENT statement is missing

File: starter/source/airbag/init_monvol.F [hunk 23]
```diff
@@ -99,7 +99,7 @@ SUBROUTINE INIT_MONVOL(T_MONVOL, T_MONVOL_METADATA,
       TYPE (SURF_)   , DIMENSION(NSURF)   :: IGRSURF
       TYPE(MONVOL_STRUCT_), DIMENSION(NVOLU), INTENT(INOUT) :: T_MONVOL
       TYPE(MONVOL_METADATA_), INTENT(INOUT) :: T_MONVOL_METADATA
-      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) :: SENSOR_TAB
+      TYPE (SENSORS_)  :: SENSORS
```
Comment (sebastienVilleneuve) [inline starter/source/airbag/init_monvol.F:102]: INTENT statement is missing

File: starter/source/boundary_conditions/ebcs/iniebcs.F [hunk 24]
```diff
@@ -68,7 +68,7 @@ SUBROUTINE INIEBCS(ALE_CONNECTIVITY, IFLAG,IGRSURF, IXS, IXQ, IXTG,
       TYPE (SURF_)   , DIMENSION(NSURF)   :: IGRSURF
       TYPE(t_ebcs_tab), TARGET, INTENT(INOUT) :: EBCS_TAB
       INTEGER,INTENT(IN) :: IFLAG, IGEO(NPROPGI,NUMGEO)
-      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) :: SENSOR_TAB
+      TYPE (SENSORS_)  :: SENSORS
```
Comment (sebastienVilleneuve) [inline starter/source/boundary_conditions/ebcs/iniebcs.F:71]: INTENT statement is missing

File: starter/source/interfaces/int07/hm_read_inter_type07.F [hunk 25]
```diff
@@ -82,7 +82,7 @@ SUBROUTINE HM_READ_INTER_TYPE07(
 C-----------------------------------------------
       TYPE (GROUP_)  ,TARGET, DIMENSION(NGRNOD)  :: IGRNOD
       TYPE (SURF_)   ,TARGET , DIMENSION(NSURF)   :: IGRSURF
-      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) :: SENSOR_TAB
+      TYPE (SENSORS_)  :: SENSORS
```
Comment (sebastienVilleneuve) [inline starter/source/interfaces/int07/hm_read_inter_type07.F:85]: INTENT statement is missing

File: starter/source/interfaces/int11/hm_read_inter_type11.F [hunk 26]
```diff
@@ -80,7 +80,7 @@ SUBROUTINE HM_READ_INTER_TYPE11(
 C-----------------------------------------------
       TYPE (SURF_)   ,TARGET , DIMENSION(NSURF)   :: IGRSURF
       TYPE (SURF_)   ,TARGET , DIMENSION(NSLIN)   :: IGRSLIN
-      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) :: SENSOR_TAB
+      TYPE (SENSORS_)  :: SENSORS
```
Comment (sebastienVilleneuve) [inline starter/source/interfaces/int11/hm_read_inter_type11.F:83]: INTENT statement is missing

File: starter/source/interfaces/int21/hm_read_inter_type21.F [hunk 27]
```diff
@@ -75,7 +75,7 @@ SUBROUTINE HM_READ_INTER_TYPE21(
       TYPE (SURF_)   ,TARGET , DIMENSION(NSURF)   :: IGRSURF
       TYPE(SUBMODEL_DATA) LSUBMODEL(*)
       TYPE (UNIT_TYPE_),INTENT(IN) ::UNITAB 
-      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) :: SENSOR_TAB
+      TYPE (SENSORS_)  :: SENSORS
```
Comment (sebastienVilleneuve) [inline starter/source/interfaces/int21/hm_read_inter_type21.F:78]: INTENT statement is missing

File: starter/source/interfaces/int24/hm_read_inter_type24.F [hunk 28]
```diff
@@ -79,7 +79,7 @@ SUBROUTINE HM_READ_INTER_TYPE24(
       TYPE(INTBUF_FRIC_STRUCT_) INTBUF_FRIC_TAB(*)
       TYPE(SUBMODEL_DATA) LSUBMODEL(*)
       TYPE (UNIT_TYPE_),INTENT(IN) ::UNITAB 
-      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) :: SENSOR_TAB
+      TYPE (SENSORS_)  :: SENSORS
```
Comment (sebastienVilleneuve) [inline starter/source/interfaces/int24/hm_read_inter_type24.F:82]: INTENT statement is missing

File: starter/source/interfaces/int25/hm_read_inter_type25.F [hunk 29]
```diff
@@ -83,7 +83,7 @@ SUBROUTINE HM_READ_INTER_TYPE25(
       TYPE(INTBUF_FRIC_STRUCT_) INTBUF_FRIC_TAB(*)
       TYPE(SUBMODEL_DATA) LSUBMODEL(*)
       TYPE (UNIT_TYPE_),INTENT(IN) ::UNITAB 
-      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) :: SENSOR_TAB
+      TYPE (SENSORS_)  :: SENSORS
```
Comment (sebastienVilleneuve) [inline starter/source/interfaces/int25/hm_read_inter_type25.F:86]: INTENT statement is missing

File: starter/source/interfaces/reader/hm_read_inter_struct.F [hunk 30]
```diff
@@ -94,7 +94,7 @@ SUBROUTINE HM_READ_INTER_STRUCT(
      .   FRIGAP(*),FRIC_P(*),I2RUPT(*),TF(*)
       CHARACTER TITR*nchartitle
       TYPE(INTBUF_FRIC_STRUCT_) INTBUF_FRIC_TAB(*)
-      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) :: SENSOR_TAB
+      TYPE (SENSORS_)  :: SENSORS
```
Comment (sebastienVilleneuve) [inline starter/source/interfaces/reader/hm_read_inter_struct.F:97]: INTENT statement is missing

File: starter/source/loads/general/load_centri/hm_read_load_centri.F [hunk 31]
```diff
@@ -79,7 +79,7 @@ SUBROUTINE HM_READ_LOAD_CENTRI(ICFIELD,LCFIELD  ,CFIELD  ,ITAB     ,ITABM1 ,
 C-----------------------------------------------
       TYPE (GROUP_)  , DIMENSION(NGRNOD)  :: IGRNOD
       TYPE(SUBMODEL_DATA),INTENT(IN)::LSUBMODEL(*)
-      TYPE (SENSOR_STR_) ,DIMENSION(NSENSOR) :: SENSOR_TAB
+      TYPE (SENSORS_)  :: SENSORS
```
Comment (sebastienVilleneuve) [inline starter/source/loads/general/load_centri/hm_read_load_centri.F:82]: INTENT statement is missing

**Review by sebastienVilleneuve — CHANGES_REQUESTED** [review]
some INTENT statement are missing
SENSORS is not passed by arguments in several routines

**Pull request closed without merging**